### PR TITLE
doc(functions): update serverless yml parameters list

### PR DESCRIPTION
**Pull request by lucasscw**
Updated the documentation to match the current system state

### Your checklist for this pull request

- [ ] Check that the commit messages match our requested structure.
- [ ] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description
Please describe your pull request.

### CODE REVIEW

**Review by bene2k1 — APPROVED** [review]

File: serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx [hunk 1]
```diff
@@ -54,19 +54,27 @@ The `serverless.yml` file contains the configuration of a namespace containing o
 
 The different parameters are:
 - `service`: your namespace name
+- `useDotenv`: load environment variables from `.env` files (default: false), read [Security and secret management](#security-and-secret-management)
+- `configValidationMode`: Configuration validation: 'error' (fatal error), 'warn' (logged to the output) or 'off' (default: warn)
 - `provider.runtime`: the runtime of your functions ([check the supported runtimes](/serverless/functions/reference-content/functions-lifecycle/))
 - `provider.env`: the environment variables attached to your namespace are injected into all your namespace containers
-- `scwToken`: the Scaleway **token** you got in prerequisites
+- `provider.secret`: secret environment variables attached to your namespace are injected to all your namespace functions, see [this example project](./examples/secrets)
+- `scwToken`: the Scaleway **API token** you got in prerequisites
```
Comment (ldecarvalho-doc) [inline serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx:62]:
```suggestion
- `scwToken`: the Scaleway **API token** you got in the requirements section
```

**Review by ldecarvalho-doc — APPROVED** [review]

File: serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx [hunk 2]
```diff
@@ -54,19 +54,27 @@ The `serverless.yml` file contains the configuration of a namespace containing o
...
   - `handler` (Required): file or function which will be executed. 
   - `env` (Optional): environment variables specifics for the current function
+  - `secret` (Optional): secret environment variables specific to the current function
   - `minScale` (Optional): how many function instances we keep running (default: 0)
   - `maxScale` (Optional): maximum number of instances this function can scale to (default: 20)
   - `memoryLimit`: RAM allocated to the function instances. 
+  - `timeout`: is the maximum duration in seconds that the request will wait to be served before it times out (default: 300 seconds)
   - `runtime`: (Optional) runtime of the function. Can be used when you need to deploy multiple functions with different runtimes in your Serverless Project. If absent, `provider.runtime` will be used.
   - `events` (Optional): List of events that trigger your functions (e.g, trigger a function based on a schedule with `CRONJobs`). 
+  - `custom_domains` (Optional): List of custom domains, refer to the [how to add a custom domain](/serverless/functions/how-to/add-a-custom-domain-name-to-a-function/) documentation page
```
Comment (RoRoJ) [inline serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx:76]:
```suggestion
  - `custom_domains` (Optional): list of custom domains, refer to the [how to add a custom domain](/serverless/functions/how-to/add-a-custom-domain-name-to-a-function/) documentation page
```

File: serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx [hunk 3]
```diff
@@ -54,19 +54,27 @@ The `serverless.yml` file contains the configuration of a namespace containing o
 
 The different parameters are:
 - `service`: your namespace name
+- `useDotenv`: load environment variables from `.env` files (default: false), read [Security and secret management](#security-and-secret-management)
+- `configValidationMode`: Configuration validation: 'error' (fatal error), 'warn' (logged to the output) or 'off' (default: warn)
```
Comment (RoRoJ) [inline serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx:58]:
```suggestion
- `configValidationMode`: configuration validation: 'error' (fatal error), 'warn' (logged to the output) or 'off' (default: warn)
```

File: serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx [hunk 4]
```diff
@@ -54,19 +54,27 @@ The `serverless.yml` file contains the configuration of a namespace containing o
...
 - `provider.runtime`: the runtime of your functions ([check the supported runtimes](/serverless/functions/reference-content/functions-lifecycle/))
 - `provider.env`: the environment variables attached to your namespace are injected into all your namespace containers
-- `scwToken`: the Scaleway **token** you got in prerequisites
+- `provider.secret`: secret environment variables attached to your namespace are injected to all your namespace functions, see [this example project](./examples/secrets)
+- `scwToken`: the Scaleway **API token** you got in prerequisites
 - `scwProject`: the Scaleway **Project ID** 
+- `scwRegion`: the Scaleway region in which the deployment will take place (default: `fr-par`)
 - `package.patterns`: usually, this parameter does not need to be configured. It allows you to include or exclude directories from the deployment
 - `functions`: Configuration of your functions. It is a `yml` dictionary, with the key being the functions' name
```
Comment (RoRoJ) [inline serverless/functions/api-cli/fun-uploading-with-serverless-framework.mdx:66]:
```suggestion
- `functions`: configuration of your functions. It is a `yml` dictionary, with the key being the functions' name
```

**Review by RoRoJ — APPROVED** [review]

**Review by glihm — COMMENTED** [review]
I leave a PR related to this one. I think one of my two commits in my PR is related to yours. But I dont agree with the `scwToken` and `scwRegion` indentation. They MUST belong to the `provider` collection, and they are not prefixed by `provider.`.

So, for me:
1. The prefix must be added.
2. To homogenise with `functions`, why `provider` is not also indented?

#1203 Here is the PR.

**Review by thomas-tacquet — APPROVED** [review]

**Comment by glihm** [comment]
Thank you for the punctuation. :)

I just wanted to explain what I mentioned in my previous comment. With a fresh pull of `serverless.yml` using the scaleway plugin, the configuration file doesn't work if the configuration keys corresponding to Scaleway credentials are not indented below `provider` key. And **the documentation is not reflecting that clearly**.

```
service: testService
configValidationMode: off
provider:
  name: scaleway
  runtime: python310 # Available python runtimes are listed in documentation
  # Global Environment variables - used in every functions
  env:
    test: test

  # <<<<< HERE, those MUST be in the provider section.
  scwToken: *****
  scwProject: ******
  scwRegion: 'region-code'

plugins:
  - serverless-scaleway-functions

# ....
```

I suggested adding the prefix `provider.`, or using the same format as the one used for the `functions` key in the documentation.
 

> The different parameters are:
> - `service`: your namespace name.
> - `useDotenv`: load environment variables from `.env` files (default: false), read [Security and secret management](#security-and-secret-management).
> - `configValidationMode`: configuration validation: 'error' (fatal error), 'warn' (logged to the output) or 'off' (default: warn).
> - `provider.runtime`: the runtime of your functions ([check the supported runtimes](/serverless/functions/reference-content/functions-lifecycle/)).
> - `provider.env`: the environment variables attached to your namespace are injected into all your namespace containers.
> - `provider.secret`: secret environment variables attached to your namespace are injected to all your namespace functions, see [this example project](./examples/secrets).
> - `provider.scwToken`: the Scaleway **API token** you got in the requirements section.
> - `provider.scwProject`: the Scaleway **Project ID**.
> - `provider.scwRegion`: the Scaleway region in which the deployment will take place (default: `fr-par`).
> - `package.patterns`: usually, this parameter does not need to be configured. It allows you to include or exclude directories from the deployment.
> - `functions`: configuration of your functions. It is a `yml` dictionary, with the key being the functions' name.
>   - `handler` (Required): file or function which will be executed. 
>   - `env` (Optional): environment variables specifics for the current function.
>   - `secret` (Optional): secret environment variables specific to the current function.
>   - `minScale` (Optional): how many function instances we keep running (default: 0).
>   - `maxScale` (Optional): maximum number of instances this function can scale to (default: 20).
>   - `memoryLimit`: RAM allocated to the function instances. 
>   - `timeout`: is the maximum duration in seconds that the request will wait to be served before it times out (default: 300 seconds).
>   - `runtime`: (Optional) runtime of the function. Can be used when you need to deploy multiple functions with different runtimes in your Serverless Project. If absent, `provider.runtime` will be used.
>   - `events` (Optional): List of events that trigger your functions (e.g, trigger a function based on a schedule with `CRONJobs`). 
>   - `custom_domains` (Optional): list of custom domains, refer to the [how to add a custom domain](/serverless/functions/how-to/add-a-custom-domain-name-to-a-function/) documentation page.
>   - `httpOption` (Optional): force https redirection, possible values are `enabled` and `redirected` (default: `enabled`).

or:

> The different parameters are:
> - `service`: your namespace name.
> - `useDotenv`: load environment variables from `.env` files (default: false), read [Security and secret management](#security-and-secret-management).
> - `configValidationMode`: configuration validation: 'error' (fatal error), 'warn' (logged to the output) or 'off' (default: warn).
> - `provider`: configuration related to Scaleway, your cloud provider.
>   - `runtime`: the runtime of your functions ([check the supported runtimes](/serverless/functions/reference-content/functions-lifecycle/)).
>   - `env`: the environment variables attached to your namespace are injected into all your namespace containers.
>   - `secret`: secret environment variables attached to your namespace are injected to all your namespace functions, see [this example project](./examples/secrets).
>   - `scwToken`: the Scaleway **API token** you got in the requirements section.
>   - `scwProject`: the Scaleway **Project ID**.
>   - `scwRegion`: the Scaleway region in which the deployment will take place (default: `fr-par`).
> - `package.patterns`: usually, this parameter does not need to be configured. It allows you to include or exclude directories from the deployment.
> - `functions`: configuration of your functions. It is a `yml` dictionary, with the key being the functions' name.
>   - `handler` (Required): file or function which will be executed. 
>   - `env` (Optional): environment variables specifics for the current function.
>   - `secret` (Optional): secret environment variables specific to the current function.
>   - `minScale` (Optional): how many function instances we keep running (default: 0).
>   - `maxScale` (Optional): maximum number of instances this function can scale to (default: 20).
>   - `memoryLimit`: RAM allocated to the function instances. 
>   - `timeout`: is the maximum duration in seconds that the request will wait to be served before it times out (default: 300 seconds).
>   - `runtime`: (Optional) runtime of the function. Can be used when you need to deploy multiple functions with different runtimes in your Serverless Project. If absent, `provider.runtime` will be used.
>   - `events` (Optional): List of events that trigger your functions (e.g, trigger a function based on a schedule with `CRONJobs`). 
>   - `custom_domains` (Optional): list of custom domains, refer to the [how to add a custom domain](/serverless/functions/how-to/add-a-custom-domain-name-to-a-function/) documentation page.
>   - `httpOption` (Optional): force https redirection, possible values are `enabled` and `redirected` (default: `enabled`).

**Pull request merged**